RecoParticleFlow/PFClusterProducerAlpaka/plugins/alpaka/PFClusterProducerKernel.dev.cc

```diff
@@ -0,0 +1,2224 @@
+
+    alpaka::memset(queue, nSeeds, 0x00);  // Reset nSeeds
+
+    // seedingTopoThresh
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        seedingTopoThreshKernel{},
+                        clusteringVars.view(),
+                        params.view(),
+                        pfRecHits.view(),
+                        pfClusters.view(),
+                        pfrhFractions.view(),
+                        nSeeds.data());
+    // prepareTopoInputs
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        prepareTopoInputsKernel{},
+                        pfRecHits.view(),
+                        clusteringVars.view(),
+                        clusteringEdgeVars.view(),
+                        nSeeds.data());
+    // ECLCC
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        eclccInitKernel{},
+                        pfRecHits.view(),
+                        clusteringVars.view(),
+                        clusteringEdgeVars.view());
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        eclccCompute1Kernel{},
+                        pfRecHits.view(),
+                        clusteringVars.view(),
+                        clusteringEdgeVars.view());
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        eclccCompute2Kernel{},
+                        pfRecHits.view(),
+                        clusteringVars.view(),
+                        clusteringEdgeVars.view());
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        eclccCompute3Kernel{},
+                        pfRecHits.view(),
+                        clusteringVars.view(),
+                        clusteringEdgeVars.view());
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(blocks, threadsPerBlock),
+                        eclccFlattenKernel{},
+                        pfRecHits.view(),
+                        clusteringVars.view(),
+                        clusteringEdgeVars.view());
+    // topoClusterContraction
+    alpaka::exec<Acc1D>(queue,
+                        make_workdiv<Acc1D>(1, threadsPerBlockForClustering),
+                        topoClusterContractionKernel{},
+                        pfRecHits.view(),
+                        clusteringVars.view(),
+                        pfClusters.view(),
+                        nSeeds.data());
+
+    // Run fillRhfIndex on serial or parallel
+    if (std::is_same_v<Device, alpaka::DevCpu>) {
+      alpaka::exec<Acc1D>(queue,
+                          make_workdiv<Acc1D>(1, 32),
```

Comment:
Why 32 elements per block in this case? I don't see `fillRhfIndexSerial()` to make any use of the element index.

(similar question for the `fastClusterSerialKernel` below)

Reply:
I think it was a general recommendation from Felice some time ago, but yes I don't think the element index is ever explicitly used.